src/Vulkan/Silk.NET.Vulkan/Vk.cs

```diff
@@ -154,34 +157,47 @@ protected override void PostInit()
         /// Checks whether the given instance extension is available.
         /// </summary>
         /// <param name="extension">The instance extension name.</param>
+        /// <param name="layer">The instance layer name.</param>
         /// <returns>Whether the instance extension is available.</returns>
-        public unsafe bool IsInstanceExtensionPresent(string extension)
+        public unsafe bool IsInstanceExtensionPresent(string extension, string layer = null)
         {
             // Note we use optimistic code to avoid locks, if this is called on multiple threads
             // then multiple initialisations can happen, only one thread will succeed in initialising the cache though.
             var cachedInstanceExtensions = _cachedInstanceExtensions;
             if (cachedInstanceExtensions is null)
             {
-                // Get count of properties
-                var instanceExtPropertiesCount = 0u;
-                EnumerateInstanceExtensionProperties((byte*) 0, &instanceExtPropertiesCount, null);
+                var layerPtr = layer == null ? 0 : SilkMarshal.StringToPtr(layer);
```

Comment:
Formalising my comment from Discord: this can be simplified by doing `fixed (byte* layerPtr = (layer is null ? Array.Empty<byte>() : Encoding.UTF8.GetBytes(layer)))`